relay-server/src/actors/envelopes.rs

```diff
@@ -148,79 +141,135 @@ impl UpstreamRequest for SendEnvelope {
     }
 }
 
-pub struct EnvelopeManager {
-    config: Arc<Config>,
-    captures: BTreeMap<EventId, CapturedEnvelope>,
-    #[cfg(feature = "processing")]
-    store_forwarder: Option<ServiceAddr<Store>>,
-    #[cfg(feature = "processing")]
-    _runtime: Runtime,
+/// Sends an envelope to the upstream or Kafka.
+#[derive(Debug)]
+pub struct SubmitEnvelope {
+    pub envelope: Envelope,
+    pub envelope_context: EnvelopeContext,
+}
+
+/// Sends a client report to the upstream.
+#[derive(Debug)]
+pub struct SendClientReports {
+    /// The client report to be sent.
+    pub client_reports: Vec<ClientReport>,
+    /// Scoping information for the client report.
+    pub scoping: Scoping,
+}
+
+/// Sends a batch of pre-aggregated metrics to the upstream or Kafka.
+///
+/// Responds with `Err` if there was an error sending some or all of the buckets, containing the
+/// failed buckets.
+#[derive(Debug)]
+pub struct SendMetrics {
+    /// The pre-aggregated metric buckets.
+    pub buckets: Vec<Bucket>,
+    /// Scoping information for the metrics.
+    pub scoping: Scoping,
+    /// The key of the logical partition to send the metrics to.
+    pub partition_key: Option<u64>,
+}
+
+/// Dispatch service for generating and submitting Envelopes.
+#[derive(Debug)]
+pub enum EnvelopeManager {
+    SubmitEnvelope(Box<SubmitEnvelope>),
+    SendClientReports(SendClientReports),
+    SendMetrics(SendMetrics, Sender<Result<(), Vec<Bucket>>>),
```

Comment:
nit: Would be nice to align these names, so either all `Submit*` or all `Send*`.

Reply:
Sounds good, will add this in a follow-up.